### DISCUSSION TITLE
Can't log in as admin after installation of "pure docker" version - "Invalid credentials"

### DISCUSSION THREAD

**Discussion by MiltonFriedman33** [photoprism/photoprism#2681] [Q&A] [answered]
I have an issue with logging into PhotoPrism as admin after installation on `Synology NAS 7.1` 
Info:
- installing `pure docker` version from this tutorial https://www.wundertech.net/how-to-setup-photoprism-on-a-synology-nas - so the database will be SQLite on the storage
- tried it with PhotoPrism docker images `latest` and `preview`
- this looks like a similar issue to https://github.com/photoprism/photoprism/issues/1103

Steps I took:
1. Downloaded docker image and configured the container
2. Admin password setup via `PHOTOPRISM_ADMIN_PASSWORD` env var to `123`. 
3. The local SQLite seems like its working, Photoprims was able to create some directories
<img width="539" alt="image" src="https://user-images.githubusercontent.com/112814189/188327772-fb2d6899-2c77-4d8b-8cd7-1e33d2b6e072.png">
4. GUI works but when I log in as admin/123 I get a message `Invalid credentials`.   

5. The logs show 2 possible issues:

- `DEBU[2022-09-04T14:40:31Z] session: open /photoprism/storage/cache/sessions.json: no such file or directory`
- a `400` error on session creation `DEBU[2022-09-04T14:42:33Z] server: POST /api/v1/session (400) [1m1.42992191s]` - the request is quite long 1m1s
  
Not sure how to debug this 400 further

**Comment by graciousgrey** [comment]
Unfortunately we don't have a Synology test device, that's why we can't reproduce cases like this or provide better set up instructions.

Please note the initial admin password must have at least 8 characters, so the invalid password might be the problem.

https://docs.photoprism.app/getting-started/config-options/

**Reply by MiltonFriedman33** [reply]
Unfortunately making the password longer (did 9 chars) didn't help. Is there a way I can debug this? The Docker log output is vague - does not say anything, can I log in to the container and find better logs?

**Reply by graciousgrey** [reply]
Did try to install it from scratch? 

Pleae note: 

_It is not possible to change the password via PHOTOPRISM_ADMIN_PASSWORD after the app has been started for the first time. You may run docker exec -ti photoprism photoprism passwd in a terminal to change an existing password. You can also reset your database for a clean start._

https://docs.photoprism.app/getting-started/docker/

**Reply by lastzero** [reply]
Make sure to use a long enough password, NOT 123 and check the logs for errors and warnings.

**Reply by MiltonFriedman33** [reply]
@graciousgrey thanks for this info - that was unexpected. I deleted the container and created a new one with a longer password and it also didn't work :(

@lastzero I used a longer (asdf12345) password. As for errors I noticed something new:
```
DEBU[2022-09-05T19:00:17Z] api-v1: abort /api/v1/config with code 401 (please log in and try again)                                 
```
is the config 401 error something that can cause login issues?

**Comment by MiltonFriedman33** [comment, marked as answer]
Problem solved, I had to:
1. ssh into Synology
2. `sudo docker ps` to get the container name
3. `sudo docker exec -ti photoprism-photoprism1 photoprism passwd` to change the password, set it to at least 6 characters

After this I was able to log in with the new password
